aggregator/election_mgr.go

```diff
@@ -504,6 +508,20 @@ func (mgr *electionManager) verifyPendingFollower(watch watch.Watch) {
 				mgr.logError("leader has not changed", errLeaderNotChanged)
 				return errLeaderNotChanged
 			}
+			_, p, err := mgr.placementManager.Placement()
+			if err != nil {
+				mgr.metrics.verifyPlacementErrors.Inc(1)
+				mgr.logError("error getting placement", err)
+				return err
+			}
+			_, exist := p.Instance(leader)
```

Comment:
Seems like this will just tell you if the host exists, but not if the `electionKey` for the which this host is the leader?

This would tell us if the leader is a valid host, but not if that host thinks it is the leader. Perhaps this might be helpful:

```
instance, exist := p.Instance(leader)
if !exists {
   // handle unknown host as leader
}
if instance.ShardSetID() != mgr.ShardSetID {
  // handle leader with different shardSetID
}

```